Interaction - Battlefield: Century of Hardship

claywar opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
* Converts Promathia mission battlefield 'Century of Hardship' to interaction framework
* Adds missing npc_list entries from captures (csnpcs not relevant, but server will give warning on battlefield finish event)
<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

## Steps to test these changes
Enter/Win/Lose/Exit Century of Hardship
<!-- Clear and detailed steps to test your changes here -->

Xaver-DaRed commented on scripts/battlefields/Mine_Shaft_2716/century_of_hardship.lua:
```diff
@@ -0,0 +1,62 @@
+                mineshaftID.mob.MOVAMUQ + 11,
+                mineshaftID.mob.MOVAMUQ + 12,
+                mineshaftID.mob.MOVAMUQ + 13,
+                mineshaftID.mob.MOVAMUQ + 14,
```
The IDs dont seem to match

MOBLIN_IDS = { 16830465, 16830466, 16830467, 16830468 }
MOBLIN_IDS = { 16830470, 16830471, 16830472, 16830473 }
MOBLIN_IDS = { 16830475, 16830476, 16830477, 16830478 }

Was the previous implementation incorrect?

claywar committed:
Interaction - Battlefield: Century of Hardship

Fix mission ID, grant XP on complete

Add Text IDs

Add missing npc_list entries from captures

Fix buggby logic for iterating moblins
